### PR TITLE
Border firelocks no longer crush mobs

### PR DESCRIPTION
**Pull request by TheChosenEvilOne**
## About The Pull Request
Stops border firelocks from crushing mobs and adds a `can_crush` variable to doors.

## Why It's Good For The Game
It makes no sense why border firelocks can crush mobs as they are border firelocks.

## Changelog
:cl:
tweak: Border firelocks no longer crush mobs.
/:cl:

### CODE REVIEW

File: code/game/machinery/doors/door.dm [hunk 1]
```diff
@@ -327,7 +328,7 @@
 	update_freelook_sight()
 	if(safe)
 		CheckForMobs()
-	else
+	else if(can_crush)
```
Comment (Cyberboss) [inline code/game/machinery/doors/door.dm]: CheckForMobs shouldn't be called either

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/machinery/doors/door.dm (hunk 1 above)
Comment (TheChosenEvilOne) [reply code/game/machinery/doors/door.dm]: Alright I just made it early return if it can't crush.

**Comment by RaveRadbury** [comment]
What are border firelocks even?

**Comment by TheChosenEvilOne** [comment]
@RaveRadbury 
image

File: code/game/machinery/doors/door.dm [hunk 2]
```diff
@@ -325,11 +326,14 @@
 	operating = FALSE
 	air_update_turf(1)
 	update_freelook_sight()
-	if(safe)
+
+	if(!can_crush)
+		return TRUE
+	else if(safe)
```
Comment (kriskog) [inline code/game/machinery/doors/door.dm]: No need for the else if you are returning early

Comment (TheChosenEvilOne) [reply code/game/machinery/doors/door.dm]: that is true.

**Pull request merged**